src/Microsoft.TestPlatform.CrossPlatEngine/TestSession/ProxyTestSessionManager.cs

```diff
@@ -96,21 +116,45 @@ public virtual bool StartSession(ITestSessionEventsHandler eventsHandler)
 
         // Wait for proxy creation to be over.
         Task.WaitAll(taskList);
+        stopwatch.Stop();
+
+        // Collecting session metrics.
+        requestData?.MetricsCollection.Add(
+            TelemetryDataConstants.TestSessionId,
+            _testSessionInfo.Id);
+        requestData?.MetricsCollection.Add(
+            TelemetryDataConstants.TestSessionSpawnedTesthostCount,
+            _proxyContainerList.Count);
+        requestData?.MetricsCollection.Add(
+            TelemetryDataConstants.TestSessionTesthostSpawnTimeInSec,
+            stopwatch.Elapsed.TotalSeconds);
```

Comment:
[minor] You will do multiple times the null check, it might be better (neglectable) to enclose the adds in a `if`.